sql/core/src/main/scala/org/apache/spark/sql/RelationalGroupedDataset.scala

```diff
@@ -416,7 +426,7 @@ class RelationalGroupedDataset protected[sql](
         new RelationalGroupedDataset(
           df,
           groupingExprs,
-          RelationalGroupedDataset.PivotType(pivotColumn.expr, values.map(Literal.apply)))
+          RelationalGroupedDataset.PivotType(pivotColumn.expr, values.map(lit(_).expr)))
```

Comment:
@MaxGekk, just for doubly doubly sure, shell we `Try(...).getOrElse(lit(...).expr)`? Looks at least there's one case of a potential behaviour change about scale and precision.

Reply:
> Looks at least there's one case of a potential behaviour change about scale and precision.

Could you explain, please. Why do you expect some behavior change?

Reply:
now we eventually call `Literal.create` instead of `Literal.apply`. I'm not sure if there is a behavior change though.

Reply:
from a quick look, seems `Literal.create` is more powerful and should not have regressions.

Reply:
That's true in general but specifically is decimal precision more correct?